routers/repo/editor.go

```diff
@@ -106,7 +106,13 @@ func editFile(ctx *context.Context, isNewFile bool) {
 			ctx.NotFound("blob.Data", err)
 			return
 		}
-		defer dataRc.Close()
+		closed := false
```

Comment:
It's unnecessary to check this because `Close` could be invoked many times, just add a line in 131 is OK.

This of course is no problem, but the code is a little ugly and not readable.

You could find many simliar codes on golang standard liberary.

Reply:
done